src/converter/one/heatmap/SymbolTable.java

```diff
@@ -0,0 +1,229 @@
+/*
+ * Copyright The async-profiler authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package one.heatmap;
+
+import java.util.Arrays;
+
+public class SymbolTable {
+
+    private static final int INITIAL_CAPACITY = 16 * 1024;
+    private static final long MARK_POST_TRANSFORM = 1L << 32;
+    private static final byte[] UNKNOWN = "<Unknown>".getBytes();
+
+    private static final byte[] BYTE = "byte".getBytes();
+    private static final byte[] CHAR = "char".getBytes();
+    private static final byte[] SHORT = "short".getBytes();
+    private static final byte[] INT = "int".getBytes();
+    private static final byte[] LONG = "long".getBytes();
+    private static final byte[] BOOLEAN = "boolean".getBytes();
+    private static final byte[] FLOAT = "float".getBytes();
+    private static final byte[] DOUBLE = "double".getBytes();
+
+    private byte[][] keys;
```

Comment:
Overhead is not tiny at all, AFAIR. I can't provide you benchmarks right now, but if I remember correctly, it was the most time consuming place (because of heavy GC pressure).

Reply:
Actually, let me re-test this.

Reply:
Re-implemented in compromised way